dexbot/strategies/relative_orders.py

```diff
@@ -118,57 +167,106 @@ def update_orders(self):
         self.clear_orders()
 
         order_ids = []
+        expected_num_orders = 0
 
         amount_base = self.amount_base
         amount_quote = self.amount_quote
 
         # Buy Side
-        buy_order = self.market_buy(amount_base, self.buy_price, True)
-        if buy_order:
-            self.save_order(buy_order)
-            order_ids.append(buy_order['id'])
+        if amount_base:
+            buy_order = self.market_buy(amount_base, self.buy_price, True)
+            if buy_order:
+                self.save_order(buy_order)
+                order_ids.append(buy_order['id'])
+            expected_num_orders += 1
 
         # Sell Side
-        sell_order = self.market_sell(amount_quote, self.sell_price, True)
-        if sell_order:
-            self.save_order(sell_order)
-            order_ids.append(sell_order['id'])
+        if amount_quote:
+            sell_order = self.market_sell(amount_quote, self.sell_price, True)
+            if sell_order:
+                self.save_order(sell_order)
+                order_ids.append(sell_order['id'])
+            expected_num_orders += 1
 
         self['order_ids'] = order_ids
 
         self.log.info("Done placing orders")
 
         # Some orders weren't successfully created, redo them
-        if len(order_ids) < 2 and not self.disabled:
+        if len(order_ids) < expected_num_orders and not self.disabled:
             self.update_orders()
 
-    def check_orders(self, *args, **kwargs):
+    def check_orders(self, event, *args, **kwargs):
         """ Tests if the orders need updating
         """
+        delta = datetime.now() - self.last_check
+
+        # Only allow to check orders whether minimal time passed
+        if delta < timedelta(seconds=self.min_check_interval):
+            self.log.debug('Ignoring market_update event as min_check_interval is not passed')
+            return
+
         orders = self.fetch_orders()
 
+        # Detect complete fill, order expiration, manual cancel, or just init
+        need_update = False
         if not orders:
-            self.update_orders()
+            need_update = True
         else:
-            orders_changed = False
-
             # Loop trough the orders and look for changes
             for order_id, order in orders.items():
                 current_order = self.get_order(order_id)
 
                 if not current_order:
-                    orders_changed = True
-                    self.write_order_log(self.worker_name, order)
+                    need_update = True
+                    self.log.debug('Could not found order on the market, it was filled, expired or cancelled')
+                    # FIXME: writing a log entry is disabled because we cannot distinguish an expired order
```

Comment:
I think it would be better to disable order expiration setting for now, so that order logging isn't totally broken

Reply:
We could also inform users that trades.csv will be useless if they use order expiration. Order expiration is a useful tool otherwise.

Reply:
If that's fine then it only needs an if statement to check if order expiration setting is on

Reply:
Suits me